src/kubeobjects/container.go

```diff
@@ -6,15 +6,23 @@ import (
 )
 
 func FindContainerInPod(pod corev1.Pod, name string) (*corev1.Container, error) {
-	for i := range pod.Spec.Containers {
-		container := &pod.Spec.Containers[i]
-		if container.Name == name {
-			return container, nil
-		}
+	container := FindContainerInPodSpec(&pod.Spec, name)
+	if container != nil {
+		return container, nil
 	}
 	podName := pod.Name
 	if podName == "" {
 		podName = pod.GenerateName
 	}
 	return nil, errors.Errorf("no container %s found for pod %s", name, podName)
 }
+
+func FindContainerInPodSpec(spec *corev1.PodSpec, name string) *corev1.Container {
```

Comment:
```suggestion
func FindContainerInPodSpec(spec *corev1.PodSpec, container string) *corev1.Container {
```
makes it clearer, but its only a nitpick